### PR TITLE
merge all the commits from the messed up iconnor_video branch

### PR DESCRIPTION
**Pull request by connortechnology**
This is a redo of my changes against feature-h264

### CODE REVIEW

**Comment by SteveGilvarry** [comment]
What did you do for this? Squash all the commits? I will compare to this but might pull the commits in separately to get all the commit messages.

**Comment by connortechnology** [comment]
There really isn't that much in here, mostly me adding some comments and messing up the spacing.  

You can wait a bit and I can try to revert the spacing/code format changes... or just merge them.

The main things are the parts where it figures out which stream is video and which is audio, that was broken, because once it foudn the video, it break'd meaning it didn't continue to look for the audio.

The second is the addition of the record_audio fields and the test for it to strip out audio.

The third thing is the generation of thumbnails from the video in includes/function.php

File: README.md [hunk 1]
```diff
@@ -3,7 +3,7 @@ ZoneMinder H264 Patch
 [![Build Status](https://travis-ci.org/ZoneMinder/ZoneMinder.png?branch=feature-h264-videostorage)](https://travis-ci.org/ZoneMinder/ZoneMinder)  [![Bountysource](https://api.bountysource.com/badge/team?team_id=204&style=bounties_received)](https://www.bountysource.com/teams/zoneminder/issues?utm_source=ZoneMinder&utm_medium=shield&utm_campaign=bounties_received)
 
 ##Feature-h264-videostorage Branch Details
-This branch supports direct recording of h264 cameras into MP4 format uisng the h264 Passthrough option, but only with FFMPEG Monitors currently. It also it provides h264 encoding from jpegs of any other monitor type and local or mpeg cameras. If you encounter any issues, please open an issue on GitHub and attach it to the h264 milestone. But do remember this is bleeding edge so it will have problems.
```
Comment (SteveGilvarry) [inline README.md:6]: Did you make this change? Or a reversion?

File: db/zm_create.sql.in [hunk 2]
```diff
@@ -344,6 +345,10 @@ CREATE TABLE `Monitors` (
   `Palette` int(10) unsigned NOT NULL default '0',
   `Orientation` enum('0','90','180','270','hori','vert') NOT NULL default '0',
   `Deinterlacing` int(10) unsigned NOT NULL default '0',
+  `SaveJPEGs` TINYINT NOT NULL DEFAULT '3' ,
```
Comment (SteveGilvarry) [inline db/zm_create.sql.in:348]: Put these into an update file yet or wait until we know what version we are going into?

File: web/api/.gitattributes [hunk 3]
```diff
@@ -1,33 +0,0 @@
-# Define the line ending behavior of the different file extensions
```
Comment (SteveGilvarry) [inline web/api/.gitattributes:1]: Where did this sneak in from?

File: web/includes/functions.php [hunk 4]
```diff
@@ -1153,30 +1153,38 @@ function zmaCheck( $monitor )
...
-    else
+    } else {
         $captImage = sprintf( "%0".ZM_EVENT_IMAGE_DIGITS."d-capture.jpg", $frame['FrameId'] );
+        if ( ! file_exists( $eventPath.'/'.$captImage ) ) {
+            # Generate the frame JPG
+            if ( $event['DefaultVideo'] ) {
+                $command ='ffmpeg -v 0 -i '.$eventPath.'/'.$Event->DefaultVideo().' -vf "select=gte(n\\,'.$frame['FrameId'].'),setpts=PTS-STARTPTS" '.$eventPath.'/'.$captImage;
+                system( $command, $output, $retval );
+            } else {
+                Error("Can't create frame images from video becuase there is no video file for this event " );
```
Comment (SteveGilvarry) [inline web/includes/functions.php]: Spelling people will get you on "becuase'

**Comment by SteveGilvarry** [comment]
Check my comments I am happy to merge with the spacing to get things in and then we can merge spacing fix, if we get that master one in also.

File: web/api/.gitignore [hunk 5]
```diff
@@ -1,21 +0,0 @@
-# User specific & automatically generated files #
```
Comment (SteveGilvarry) [inline web/api/.gitignore:1]: Same on this one?

**Comment by connortechnology** [comment]
We have decided that this will go into 1.30.  If we don't make it for some reason, we can rename the update file to 1.31.

File: README.md (hunk 1 above)
Comment (connortechnology) [inline README.md:6]: Not intentionally, but I think some updates here are in order.

**Pull request merged**